Discovery/Discovery/Presentation/WebPrograms/ProgramWebviewViewModel.swift

```diff
@@ -94,6 +100,27 @@ public class ProgramWebviewViewModel: ObservableObject {
             return .enrollOpen
         }
     }
+    
+    @MainActor
+    func updateCookies(force: Bool = false) async {
```

Comment:
it's same method as in WebUnitViewModel better to move it to one place to be reusable if it will need in other place. Protocol for example.

Reply:
@forgotvas Underneath, they are using the shared code, which is `authInteractor.getCookies(force: force)` and this method is just for handling the errors etc, do we still need to move this method to a common place?

Reply:
So this is what you meant when you said to move it to common place, or you have something else on mind?

```
public protocol UpdateCookies {
    var updatingCookies: Bool { get }
    var errorMessage: String? { get }
    var retryCount: Int { get }
    func updateCookies(force: Bool) async
}

public class UpdateCookiesModel: UpdateCookies {
    @Published public var updatingCookies: Bool
    @State public var errorMessage: String?
    public var retryCount: Int
    let authInteractor: AuthInteractorProtocol
    
    init(updatingCookies: Bool, errorMessage: String?, retryCount: Int, authInteractor: AuthInteractorProtocol) {
        self.updatingCookies = updatingCookies
        self.errorMessage = errorMessage
        self.authInteractor = authInteractor
        self.retryCount = retryCount
    }
    
    public func updateCookies(force: Bool) async {
        guard !updatingCookies else { return }
        do {
            updatingCookies = true
            try await authInteractor.getCookies(force: force)
            updatingCookies = false
            errorMessage = nil
        } catch {
            if error.isInternetError {
                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
            } else if retryCount > 0 {
                retryCount -= 1
                await updateCookies(force: force)
            } else {
                errorMessage = CoreLocalization.Error.unknownError
            }
            updatingCookies = false
        }
    }
}

Reply:
No, in my mind i see 2 options:
1) **Protocols**. Protcol that contains function and it doesn't to be described twice, only required variables: https://gist.github.com/forgotvas/ad4980a16731127fb7ddde37746000eb
2) **Inheritance**. In that case we are using base ObservableObject that contains properties and method: https://gist.github.com/forgotvas/c4330000d3dc22de514fec1c64543d4a